src/Microsoft.ML.AutoML/API/ExperimentSettings.cs

```diff
@@ -70,6 +70,18 @@ public ExperimentSettings()
             CacheBeforeTrainer = CacheBeforeTrainer.Auto;
             MaxModels = int.MaxValue;
         }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ExperimentSettings"/> using the temp file location provided to the MLContext.
+        /// </summary>
+        public ExperimentSettings(MLContext context)
```

Comment:
I'm not really excited about this now that I see it. I didn't understand the issue previously. Mostly, I don't like the duplication, and how many things needed to change.

Note that AutoML is still an "experimental" library, which means we can make breaking changes to the API, if we believe they are the right changes going forward.

I can think of a couple alternatives:

1. Don't make any changes to AutoML - keep the `CacheDirectory` property defaulting to `Path.GetTempPath()`, and not respecting `MLContext.TempFilePath`. The user has full control, but they just need to set 2 different properties.
2. Remove `CacheDirectory` from ExperiementSettings, and just use `context.TempFilePath` where `CacheDirectory` was used before.
2.b. Rename `CacheDirectory` to just be the directory name (i.e. `CacheDirectoryName`) and create that directory under `MLContext.TempFilePath`.

Thoughts?

Reply:
I think option 2b is the best. AutoML is just supposed to help figure out pipelines automatically, I think it still makes sense to keep a setting like that inside of the MLContext itself. I do think having the option of a separate folder though is good because if that value is set then AutoML will write all the models to disk, so it makes sense to have a folder just for them.

Reply:
That sounds good to me. Thanks!

Reply:
Alright, change 2b has been implemented and pushed. Take a look and let me know if its what you were thinking.